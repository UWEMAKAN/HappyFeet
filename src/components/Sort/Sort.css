.Sort {
  padding: 2rem 0 0 2rem;
  display: flex;
  align-items: center;
  border-bottom: 1px solid #dedede;
}

.Sort h2 {
  font-size: 1.6rem;
  font-weight: 600;
}

.Sort button {
  outline: none;
  background-color: #fff;
  margin-left: 3rem;
  cursor: pointer;
  font-size: 1.6rem;
  border: 1px solid transparent;
  padding: 1rem;
}

.Sort button:hover,
.Sort button:active {
  background-color: #dedede;
  border: 1px solid #adadad;
}

@media screen and (max-width: 900px) {
  .Sort button {
    margin-left: 1rem;
    padding: .5rem;
  }
}

@media screen and (max-width: 600px) {
  .Sort {
    border: none;
  }
}
