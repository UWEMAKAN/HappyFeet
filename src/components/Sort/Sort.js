import React from 'react';
import propTypes from 'prop-types';
import './Sort.css';

const Sort = ({ sortHandler }) => (
  <div className="Sort">
    <h2>Sort By</h2>
    <button onClick={sortHandler} name="" tabIndex="-3" type="button">Relevance</button>
    <button onClick={sortHandler} name="ascending" tabIndex="-2" type="button">Price - Low to High</button>
    <button onClick={sortHandler} name="descending" tabIndex="-1" type="button">Price - High to Low</button>
  </div>
);

Sort.propTypes = {
  sortHandler: propTypes.func.isRequired
};

export default Sort;
