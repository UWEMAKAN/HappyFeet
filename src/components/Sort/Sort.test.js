import React from 'react';
import { shallow } from 'enzyme';
import Sort from './Sort';

const sortHandler = jest.fn();
const renderSort = () => shallow(<Sort sortHandler={sortHandler} />);

describe('Sort Component', () => {
  test('renders Sort component', () => {
    const wrapper = renderSort();
    expect.assertions(6);
    expect(wrapper.find('h2').text()).toEqual('Sort By');
    expect(wrapper.find('button').length).toBe(3);
    expect(wrapper.find('button').first().text()).toEqual('Relevance');
    expect(wrapper.find('button').at(1).text()).toEqual('Price - Low to High');
    expect(wrapper.find('button').last().text()).toEqual('Price - High to Low');
    expect(wrapper).toMatchSnapshot();
  });

  test('should call eventHandlers', () => {
    const wrapper = renderSort();
    expect.assertions();
    wrapper.find('button').first().simulate('click');
    wrapper.find('button').at(1).simulate('click');
    wrapper.find('button').last().simulate('click');
    expect(sortHandler).toHaveBeenCalledTimes(3);
  });
});
