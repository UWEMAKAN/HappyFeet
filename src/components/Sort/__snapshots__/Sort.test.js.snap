// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort Component renders Sort component 1`] = `ShallowWrapper {}`;
